### PR TITLE
Fix loading multiple draco models wit the same url

### LINKED ISSUE #6668: Unable to load two Draco models with the same url
See this [Sandcastle]( https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=1VVRb9MwEP4rVl+aoOB0XcfEKIPRCZBgMK0VD1AevOTaWHPsyHbaFbT/ztlO26Qbr0hUVVU79919/u47Z8U0WXFYgyaviYQ1mYDhdUm/+b2on/nlREnLuATdj1/N5QoxJgMJCAlY6pf4aC4XtcwsV5JkGpiFK5WDiGotElIAXxY2IZrJXJUTlhXwCTYx+T2XBD/hMaZsGOSwYLWw35ioIdpiB3TgGLh4x0JpvuRyj5kwbfEfk8d0oVV5CUsNYKLnR8NjOjgdjV4cvUzIaEQHJ4Pj08GLLad2ytIRvmJW8/t93hlyNgulS0MLYDmXy2tus+JGCTFT7/k95O81KyEKfJK2kB8P4qM49jp16mElryCtNC+55SswlOV51OTwIvoTfRB2ETWCuQ8KS87cb7Lfa5/grL1qxWSN+hjQbQd5QzC4oGE3il12ia3A5ucB/hBv5Qq/vgDFZueba2TIDVBbgIy2Roh8wK7N23NnqJdmeweF9auDIPSdRtG8O4NCBn0FclqxDCYeMtnFHIAdZkgH5Fk4UcnuAxV6q/BI2JNpVYAGPGrOa5M0jFDm2lgUXQLTcSvlngstueRlXX5XaDFuLJOZGwWNlQb05PAIIK2nv+2lb8WIluhtXonNu8214tI21NrdIk+yDfk6Ftu7Porjg/KNXTv1C2rVDW6jp6PhcWuktqDKmfWvkOfDA0gjnFDq7sJGTxDszACTSzfPficJpRKvHfZqR/8hpgpdpNdop72TQGul45aR1hxduqYMm2Kbp7sE+OfBDZq/JyqHN3ikHw3awr1Fb/dvvM/DPJA72PSbIcFoEJC5mF35jocfzauGUq3gwk14W5vWLdinNN19L6rKpFNWVgIumWWpb7ZJLzXL1ESVlQZjIE+Dgldc3M10nd3RJV4AfbwHE2J1Df+g0km3Ekr6kJADEb+o/1HABRPmHynYKeUk/OnfAVM0X8aMFeBu+5lS4pbpK5B11BjWQXpJb2zsRsB5gL/lZaW0dZd+hNwsIC2kbdJbrAmWZsbDxukWNM75ivD89bx38DKf90gmmDH4ZFELMeW/YN47H6cY34EJ5Yf16wq0YBsXUhydfw6blNJxisvHKBuO08r4Bw).

The bug seems to be caused by two Draco models sharing the same cached resources. Maybe the cached resources are not getting set correctly after the first model is loaded?

This is also impacting i3dm tilesets where tiles point to the same external glTF.

@ggetz can you look into this?

### PR DESCRIPTION
**Pull request by ggetz**
Fixes #6668 by caching decoded Draco data for models with the same cache key (url).

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@ggetz, thanks for the pull request! Maintainers, we have a signed CLA from @ggetz, so you can review this at any time.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Specs/Scene/ModelSpec.js [hunk 1]
```diff
@@ -2380,14 +2380,14 @@ defineSuite([
         });
     });
 
-    it('loads a gltf with WEB3D_quantized_attributes SCALAR attribute', function() {
+    fit('loads a gltf with WEB3D_quantized_attributes SCALAR attribute', function() {
```
Comment (mramato) [inline Specs/Scene/ModelSpec.js]: you left a couple of `fit` in the code.

**Comment by mramato** [comment]
I'll let @lilleyse review and merge, but I can confirm that this fixes the original issue I was seeing with draco encoded i3dm tiles.  Thanks @ggetz!

File: Source/Scene/DracoLoader.js [hunk 2]
```diff
@@ -139,18 +139,38 @@ define([
...
-    DracoLoader.parse = function(model) {
+    DracoLoader.parse = function(model, context) {
         if (!DracoLoader.hasExtension(model)) {
             return;
         }
 
         var loadResources = model._loadResources;
+        var cacheKey = model.cacheKey;
+        if (defined(cacheKey)) {
+            if (!define(DracoLoader._decodedModelResourceCache)) {
```
Comment (lilleyse) [inline Source/Scene/DracoLoader.js]: Should be `defined`

File: Source/Scene/Model.js [hunk 3]
```diff
@@ -4585,6 +4587,7 @@ define([
 
         this._rendererResources = undefined;
         this._cachedRendererResources = this._cachedRendererResources && this._cachedRendererResources.release();
+        DracoLoader.destroyCachedDataForModel(this);
```
Comment (lilleyse) [inline Source/Scene/Model.js]: Should the cached data be reference counted, similar to `CachedRendererResources`? This scenario isn't working for me:

1. Load first model
2. Load second model
3. Destroy first model
4. Load third model

[localhost:8080 Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html#c=vVZdb9s2FP0rhF8sDSqlOE6LNW621EE7YA0WxEEfBr8w4rVFhCIFkrLjDf3vvSQlW/baLdhDBUMQqftx7rnnUt4wQzYCtmDIO6JgS+ZgRVvTz2EvGZdhOdfKMaHAjNPLpdqgjy1BAbpEXxqW+GqpVq0qndCKlAaYg1vNQSatkRmpQKwrlxHDFNf1nJUV/A67lPy9VASv+BpDdgg4rFgr3WcmW0h634IWHoG39yi0EWuhDj5zZhw+MXVOV0bXN7A2ADZ5dTY5p8Wb6fT12c8ZmU5pcVGcvyle95iGIWsP+JY5I54PcR8Qs11pU1taAeNCre+EK6t7LeWD/iCegX8wrIYk4smGRP52Yp+kaeDpKB9mCgzSxohaOLEBSxnnSRcjkBgq+ijdKukI8xcSS976e3bYG1bwdrga2JQd+2hw3A7yC0HjisbdJPXRFbYCm8+j+5e0pyveDbjWqJiHYs/57g6BCgvUVaCSXg9JMNh3uy+/RNoMOwgpri9PjFB+BrkLIo1EWZQXqEXDSpgHl/ne5sTZ+0xoQX6KhdXsOUKhjxorw9YsmgoMYMVctDbrECHbrXXIvQJm0kHIAxZaCyXqtv5To9KEdUyVfiIMZiroxWkJoFyA37c0dGRKa5S4aOTu/e5OC+U6aMOmkW+ijfGOlHYQf5KmJ+k71R7lr6jT97iN0k4m54PJ6p0ar9nvuryanLh0xEmtn65d8g2AR6PA1NqPddjJYqoscIe9OoI/1NdlL0GqUVxmiyo7CAyM0SYd6GsrUMNbyrBXrnu7D4APX/wYxtNsJYx1/ckG2GLeLVwlDA/zusCBKJl1EvxkPmgtH5l53zqHmcfXnMcY44zs8ey1PjwJx5Tm+99109h8wepGwg1zLA812vzGsFLPdd0YsBZ4Hum7FfLpwbTlE13jIYCJCszFpIX0PwctQMNGHnPoKUhfVlvk5EcWN6EXLy4vovu/9d1DrTfwb+3Tymp0lXqdBKv952Lw7Rue3CZEPNi+lOUgth9J8gV9uYYCuO9wPMpGM+t2Eq7i/q+ibrRx/ruUIEgHiA/x2/wRk4OjpbXec5b3TjMuNkTwd8vRyf+N5YiUklmLb1atlAvxFyxHV7Mc7Y/cpA4HyR8bMJLtvEl1dvUpblJKZzku/+nlYg8GEb8C)

File: CHANGES.md [hunk 4]
```diff
@@ -9,6 +9,7 @@ Change Log
 
 ##### Fixes :wrench:
 * Fixed a bug causing crashes with custom vertex attributes on `Geometry` crossing the IDL. Attributes will be barycentrically interpolated. [#6644](https://github.com/AnalyticalGraphicsInc/cesium/pull/6644)
+* Fixed a bug with loading two Draco models with the same url. [#6668](https://github.com/AnalyticalGraphicsInc/cesium/issues/6668)
```
Comment (mramato) [inline CHANGES.md]: Maybe mention that this also applies to draco encoded i3dm tiles, since that's where most users have run into it.

**Comment by ggetz** [comment]
@lilleyse Fixed issue with https://github.com/AnalyticalGraphicsInc/cesium/pull/6673#discussion_r194451628, ready for another look.

**Comment by lilleyse** [comment]
The updates look good, thanks @ggetz!

**Pull request merged**